### ISSUE TITLE
geom_flame() in plotly

### ISSUE THREAD

**Issue by robwschlegel** [ropensci/plotly#1452]
---
Hello,
While putting a shiny app together I only now realised that ggplot geoms must first be implemented into plotly for them to work through ggploty() if they are not part of an already recognised suite of functions.

As you may know, geom_ribbon() does not fill in gaps in corners, nor does it natively allow for areas only above or below a chosen line to be filled. This is a known issue on GitHub that was never fixed:
https://github.com/tidyverse/ggplot2/issues/1642 
So I wrote a function that does all of these things:
heatwaveR::geom_flame()

I also e-mailed the author of the ggplot2 package about this to see if he would like to include it in ggplot2.

But in the meantime I was wondering if it would be possible to have heatwaveR::geom_flame() implemented in plotly. 

All the best,
-Robert Schlegel

```r
# Packages
require(heatwaveR)
require(plotly)

# Time series
ts_res <- heatwaveR::ts2clm(data = heatwaveR::sst_WA, 
                            climatologyPeriod = c("1982-01-01", "2011-12-31"))
ts_res_sub <- ts_res[10500:10800,]

# Figure
p <- ggplot(data = ts_res_sub, aes(x = t)) +
  heatwaveR::geom_flame(aes(y = temp, y2 = thresh), fill = "salmon") +
  geom_line(aes(y = temp)) +
  geom_line(aes(y = seas), colour = "green") +
  geom_line(aes(y = thresh), colour = "red") +
  labs(x = "", y = "Temperature (°C)")
ggplotly(p)
```

**Comment by cpsievert** [comment]
Cool, thanks! After a quick look, it looks like we might be able to just define `to_basic.GeomFlame()` and re-use `geom2trace.GeomPolygon()`, similar to how I've outlined here -- https://plotly-book.cpsievert.me/translating-custom-ggplot2-geoms.html

I likely won't have the bandwidth to maintain this geom's `ggplotly()` hook, but I'd be happy to help you implement and maintain this in the heatwaveR package...

BTW, here is the implementation of ggalt's converters -- I'm thinking we do something similar in your package https://github.com/hrbrmstr/ggalt/blob/master/R/geom2plotly.r

**Comment by robwschlegel** [comment]
Hello,
Thank you very much for the feedback. I will look into making these changes on my side as soon as I have a free moment. Perhaps within a week.
Once I sort this out I will close the issue. Or for your housekeeping would you prefer I just close the issue now?
All the best,
-Robert

**Comment by cpsievert** [comment]
Feel free to leave it open and report in on your progress

**Comment by robwschlegel** [comment]
Hello,
I tried your recommendations and they didn't quite work because GeomFlame cuts the geom up into a few parts, which then wants to return a data.frame that is larger than the input data. geom2trace.GeomPolygon wasn't happy with that so I went on a deep dive through the plotly source code to get a better understanding of just what was going on. Sjoe! What an amazing collection of code.

After I pieced out how it is that geom2trace expects to receive data I basically just copied in the code from GeomFlame and put that in between calling geom2trace.GeomFlame and calling  geom2trace.GeomPolygon. The source code may be found [here](https://github.com/robwschlegel/heatwaveR/blob/master/R/geom2plotly.R) if you're interested.

TLDR; I got it to work. 

Below is the reprex I sent in originally, except now it works nicely.

# Packages
require(heatwaveR)
require(plotly)

# Time series
ts_res <- heatwaveR::ts2clm(data = heatwaveR::sst_WA,
                            climatologyPeriod = c("1982-01-01", "2011-12-31"))
ts_res_sub <- ts_res[10500:10800,]

# Flame Figure
p <- ggplot(data = ts_res_sub, aes(x = t, y = temp)) +
  heatwaveR::geom_flame(aes(y2 = thresh), fill = "salmon") +
  geom_line(aes(y = temp)) +
  geom_line(aes(y = seas), colour = "green") +
  geom_line(aes(y = thresh), colour = "red") +
  labs(x = "", y = "Temperature (°C)")
p
ggplotly(p)

**Comment by cpsievert** [comment]
Wow, nice work!! 🍻
